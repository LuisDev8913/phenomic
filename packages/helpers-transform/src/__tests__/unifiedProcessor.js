// @flow

import unifiedProcessor from "../unifiedProcessor.js";

// ripped from plugin-transform-markdown
const defaultMdOptions = {
  output: "html",
  plugins: [
    require("remark-parse"),
    require("remark-toc"),
    require("remark-highlights"),
    [require("remark-rehype"), { allowDangerousHTML: true }],
    require("rehype-raw"),
    require("rehype-slug"),
    require("rehype-autolink-headings"),
  ],
};

const input = `## Test
[link](href)
\`\`\`js
console.log(window)
\`\`\`

<button>Random Html</button>
<Button>Random react like stuff</Button>

| Left-aligned | Center-aligned | Right-aligned |
| :---         |     :---:      |          ---: |
| left one     | center one     | right one     |
| left two     | center two     | right two     |
`;

it("should render markdown as json (react component)", () => {
  const processor = unifiedProcessor({
    output: "json",
    plugins: defaultMdOptions.plugins,
  });
  // $FlowFixMe mixed insn't ok?
  const md = processor.processSync(input).contents;
  expect(md).toMatchSnapshot();
});

it("should render markdown as html", () => {
  const processor = unifiedProcessor({ plugins: defaultMdOptions.plugins });
  // $FlowFixMe mixed insn't ok?
  const md = processor.processSync(input).contents;
  expect(md).toMatchSnapshot();
});
