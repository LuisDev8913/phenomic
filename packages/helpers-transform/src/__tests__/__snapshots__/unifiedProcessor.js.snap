// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render markdown as html 1`] = `
"<h2 id=\\"test\\"><a aria-hidden=\\"true\\" href=\\"#test\\"><span class=\\"icon icon-link\\"></span></a>Test</h2>
<p><a href=\\"href\\">link</a></p>
<pre class=\\"editor editor-colors\\"><div class=\\"line\\"><span class=\\"syntax--source syntax--js\\"><span class=\\"syntax--entity syntax--name syntax--type syntax--object syntax--console syntax--js\\"><span>console</span></span><span class=\\"syntax--meta syntax--method-call syntax--js\\"><span class=\\"syntax--meta syntax--delimiter syntax--method syntax--period syntax--js\\"><span>.</span></span><span class=\\"syntax--support syntax--function syntax--console syntax--js\\"><span>log</span></span><span class=\\"syntax--meta syntax--arguments syntax--js\\"><span class=\\"syntax--punctuation syntax--definition syntax--arguments syntax--begin syntax--bracket syntax--round syntax--js\\"><span>(</span></span><span class=\\"syntax--support syntax--variable syntax--dom syntax--js\\"><span>window</span></span><span class=\\"syntax--punctuation syntax--definition syntax--arguments syntax--end syntax--bracket syntax--round syntax--js\\"><span>)</span></span></span></span></span></div></pre>
<p><button>Random Html</button>
<button>Random react like stuff</button></p>




















<table><thead><tr><th align=\\"left\\">Left-aligned</th><th align=\\"center\\">Center-aligned</th><th align=\\"right\\">Right-aligned</th></tr></thead><tbody><tr><td align=\\"left\\">left one</td><td align=\\"center\\">center one</td><td align=\\"right\\">right one</td></tr><tr><td align=\\"left\\">left two</td><td align=\\"center\\">center two</td><td align=\\"right\\">right two</td></tr></tbody></table>"
`;

exports[`should render markdown as json (react component) 1`] = `
Object {
  "c": Array [
    Object {
      "c": Array [
        Object {
          "c": Array [
            Object {
              "p": Object {
                "class": "icon icon-link",
              },
              "t": "span",
            },
          ],
          "p": Object {
            "aria-hidden": "true",
            "href": "#test",
          },
          "t": "a",
        },
        "Test",
      ],
      "p": Object {
        "id": "test",
      },
      "t": "h2",
    },
    "
",
    Object {
      "c": Array [
        Object {
          "c": Array [
            "link",
          ],
          "p": Object {
            "href": "href",
          },
          "t": "a",
        },
      ],
      "t": "p",
    },
    "
",
    Object {
      "c": Array [
        Object {
          "c": Array [
            Object {
              "c": Array [
                Object {
                  "c": Array [
                    Object {
                      "c": Array [
                        "console",
                      ],
                      "t": "span",
                    },
                  ],
                  "p": Object {
                    "class": "syntax--entity syntax--name syntax--type syntax--object syntax--console syntax--js",
                  },
                  "t": "span",
                },
                Object {
                  "c": Array [
                    Object {
                      "c": Array [
                        Object {
                          "c": Array [
                            ".",
                          ],
                          "t": "span",
                        },
                      ],
                      "p": Object {
                        "class": "syntax--meta syntax--delimiter syntax--method syntax--period syntax--js",
                      },
                      "t": "span",
                    },
                    Object {
                      "c": Array [
                        Object {
                          "c": Array [
                            "log",
                          ],
                          "t": "span",
                        },
                      ],
                      "p": Object {
                        "class": "syntax--support syntax--function syntax--console syntax--js",
                      },
                      "t": "span",
                    },
                    Object {
                      "c": Array [
                        Object {
                          "c": Array [
                            Object {
                              "c": Array [
                                "(",
                              ],
                              "t": "span",
                            },
                          ],
                          "p": Object {
                            "class": "syntax--punctuation syntax--definition syntax--arguments syntax--begin syntax--bracket syntax--round syntax--js",
                          },
                          "t": "span",
                        },
                        Object {
                          "c": Array [
                            Object {
                              "c": Array [
                                "window",
                              ],
                              "t": "span",
                            },
                          ],
                          "p": Object {
                            "class": "syntax--support syntax--variable syntax--dom syntax--js",
                          },
                          "t": "span",
                        },
                        Object {
                          "c": Array [
                            Object {
                              "c": Array [
                                ")",
                              ],
                              "t": "span",
                            },
                          ],
                          "p": Object {
                            "class": "syntax--punctuation syntax--definition syntax--arguments syntax--end syntax--bracket syntax--round syntax--js",
                          },
                          "t": "span",
                        },
                      ],
                      "p": Object {
                        "class": "syntax--meta syntax--arguments syntax--js",
                      },
                      "t": "span",
                    },
                  ],
                  "p": Object {
                    "class": "syntax--meta syntax--method-call syntax--js",
                  },
                  "t": "span",
                },
              ],
              "p": Object {
                "class": "syntax--source syntax--js",
              },
              "t": "span",
            },
          ],
          "p": Object {
            "class": "line",
          },
          "t": "div",
        },
      ],
      "p": Object {
        "class": "editor editor-colors",
      },
      "t": "pre",
    },
    "
",
    Object {
      "c": Array [
        Object {
          "c": Array [
            "Random Html",
          ],
          "t": "button",
        },
        "
",
        Object {
          "c": Array [
            "Random react like stuff",
          ],
          "t": "button",
        },
      ],
      "t": "p",
    },
    "




















",
    Object {
      "c": Array [
        Object {
          "c": Array [
            Object {
              "c": Array [
                Object {
                  "c": Array [
                    "Left-aligned",
                  ],
                  "p": Object {
                    "align": "left",
                  },
                  "t": "th",
                },
                Object {
                  "c": Array [
                    "Center-aligned",
                  ],
                  "p": Object {
                    "align": "center",
                  },
                  "t": "th",
                },
                Object {
                  "c": Array [
                    "Right-aligned",
                  ],
                  "p": Object {
                    "align": "right",
                  },
                  "t": "th",
                },
              ],
              "t": "tr",
            },
          ],
          "t": "thead",
        },
        Object {
          "c": Array [
            Object {
              "c": Array [
                Object {
                  "c": Array [
                    "left one",
                  ],
                  "p": Object {
                    "align": "left",
                  },
                  "t": "td",
                },
                Object {
                  "c": Array [
                    "center one",
                  ],
                  "p": Object {
                    "align": "center",
                  },
                  "t": "td",
                },
                Object {
                  "c": Array [
                    "right one",
                  ],
                  "p": Object {
                    "align": "right",
                  },
                  "t": "td",
                },
              ],
              "t": "tr",
            },
            Object {
              "c": Array [
                Object {
                  "c": Array [
                    "left two",
                  ],
                  "p": Object {
                    "align": "left",
                  },
                  "t": "td",
                },
                Object {
                  "c": Array [
                    "center two",
                  ],
                  "p": Object {
                    "align": "center",
                  },
                  "t": "td",
                },
                Object {
                  "c": Array [
                    "right two",
                  ],
                  "p": Object {
                    "align": "right",
                  },
                  "t": "td",
                },
              ],
              "t": "tr",
            },
          ],
          "t": "tbody",
        },
      ],
      "t": "table",
    },
  ],
  "t": "div",
}
`;
